ShardedRedisClient.js

```diff
@@ -280,59 +280,32 @@ SHARDABLE.forEach((cmd) => {
 
     mainCb = _.once(mainCb);
 
-    const isReadCmd = READ_ONLY.indexOf(cmd) !== -1;
-    const timeout = isReadCmd ? this._readTimeout : this._writeTimeout;
-
     const _this = this;
-    let timeoutHandler = null;
     let breaker = client._breaker;
 
-    const timeoutCb = args[args.length - 1] = function (err) {
-      if (timeoutHandler)
-        clearTimeout(timeoutHandler);
-
-      if (!err) {
-        if (breaker)
-          breaker.pass();
-
-        return mainCb.apply(this, arguments);
-      }
-
-      // For now, both emit and log. Eventually, logging will be removed
-      _this.emit('err', new Error(`sharded-redis-client [${client.address}] err: ${err}`));
-      console.error(new Date().toISOString(), `sharded-redis-client [${client.address }] err: ${err}`);
-
-      if (breaker && err.message !== 'breaker open')
-        breaker.fail();
-
-      if (!client._isMaster) {
-        client = wrappedClient.slaves.next(client);
+    execute();
 
-        if (client._rrindex == startIndex)
-          client = findMasterClient(shardKey, _this._wrappedClients);
+    function execute() {
+      breaker.exec(cmd, args)
+        .then(result => mainCb(null, result))
+        .catch((err) => {
+          _this.emit('err', new Error(`sharded-redis-client [${client.address}] err: ${err}`));
+          console.error(new Date().toISOString(), `sharded-redis-client [${client.address }] err: ${err}`);
 
-        breaker = client._breaker;
+          if(!client._isMaster) {
+            client = wrappedClient.slaves.next(client);
 
-        return wrappedCmd(client, args);
-      }
-
-      mainCb.apply(this, arguments);
-    };
-
-    wrappedCmd(client, args);
+            if (client._rrindex == startIndex)
```

Comment:
Any reason for this double equal instead of a triple equal?

Reply:
this was the old code, i just copied it. i guess it could be triple equal

Reply:
Yea I saw that after commenting. Probably better to switch to triple equal if we don't explicitly want to allow for anything other than the expected value.